src/WebJobs.Script/Rpc/LanguageWorkerChannelInitializationService.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.Azure.WebJobs.Script.Rpc
+{
+    public class LanguageWorkerChannelInitializationService : IHostedService
+    {
+        private readonly IEnumerable<string> _languages = new List<string>()
+        {
+            LanguageWorkerConstants.JavaLanguageWorkerName
+        };
+
+        private readonly string _workerRuntime;
+
+        private ILanguageWorkerChannelManager _languageWorkerChannelManager;
+
+        public LanguageWorkerChannelInitializationService(ILanguageWorkerChannelManager languageWorkerChannelManager)
+        {
+            _languageWorkerChannelManager = languageWorkerChannelManager ?? throw new ArgumentNullException(nameof(languageWorkerChannelManager));
+            _workerRuntime = SystemEnvironment.Instance.GetEnvironmentVariable(LanguageWorkerConstants.FunctionWorkerRuntimeSettingName);
+        }
+
+        public Task StartAsync(CancellationToken cancellationToken)
+        {
+            if (SystemEnvironment.Instance.IsLinuxAppServiceEnvironment())
+            {
+                return Task.CompletedTask;
+            }
+            if (SystemEnvironment.Instance.IsLinuxContainerEnvironment())
+            {
+                return Task.CompletedTask;
+            }
+            if (string.IsNullOrEmpty(_workerRuntime))
+            {
+                return Task.WhenAll(_languages.Select(l => _languageWorkerChannelManager.InitializeChannelAsync(l)));
```

Comment:
Can we make this explicit by doing an environment.IsPlaceholderModeEnabled check? That way we're not accidentally spinning all these up if the worker runtime isn't set for some reason. We only want to do this when in placeholder mode, right?

Reply:
@pragnagopa I remember the placeholder mode check being one of the things we had discussed for this logic. I agree that it wouldn't make sense to start the worker in those cases if the selected runtime doesn't match

Reply:
@fabiocav - Yes. If the WorkerRuntime is set, worker channel matching that will be started. If workerRuntime is not set, then language workers whitelisted here start, for now this only java
@mathewc  - This change is not specialized to place holder mode. We want to move all the language workers out of jobhost.